pos-print/src/main/java/com/clouway/pos/print/adapter/http/PrintService.kt

```diff
@@ -23,23 +23,32 @@ class PrintService @Inject constructor(private var factory: PrinterFactory) {
 
   @Post
   fun printReceipt(request: Request): Reply<*> {
+    val status: Set<String>
     try {
       val dto = request.read(ReceiptDTO::class.java).`as`(GsonTransport::class.java)
       val printer = factory.getPrinter(dto.sourceIp)
       when {
         dto.fiscal
-        -> printer.printFiscalReceipt(dto.receipt)
+        -> status = printer.printFiscalReceipt(dto.receipt)
         else
-        -> printer.printReceipt(dto.receipt)
+        -> status = printer.printReceipt(dto.receipt)
       }
       printer.close()
     } catch (e: DeviceNotFoundException) {
       return Reply.with(ErrorResponse("Device not found.")).badRequest()
     } catch (e: IOException) {
       return Reply.with(ErrorResponse("Device can't connect.")).status(480)
     }
-    return Reply.with("OK")
+
+    val response = adapt(status)
+    return Reply.with(response).`as`(GsonTransport::class.java)
+  }
+
+  private fun adapt(statuses: Set<String>): PrintReceiptResponse {
```

Comment:
This method is not required and I'm not going to give any suggestion as my previous comment is going to change things here.